docs/user/common.rst

```diff
@@ -0,0 +1,52 @@
+.. _common:
+
+**************************
+Common Issues
+**************************
+
+Pandas support for ``ExtensionArray`` is still in development. As a result, there are some common issues that pint-pandas users may encounter.
+This page provides some guidance on how to resolve these issues.
+
+Units in Cells (Object dtype columns)
+-------------------------------------
+
+The most common issue pint-pandas users encouter is that they have a DataFrame with column that aren't PintArrays.
+An obvious indicator is unit strings showing in cells when viewing the DataFrame.
+Several pandas operations return numpy arrays of ``Quantity`` objects, which can cause this.
```

Comment:
Quantity objects within Pandas DataFrames (or Series) will behave like Quantities, meaning that they are subject to unit conversion rules and will raise errors when incompatible units are mixed.  But these loose Quantities don't offer the elegance or performance optimizations that come from using PintArrays.  And they may give strange error messages as Pandas tries to convert incompatible units to dimensionless magnitudes (which is often prohibited by Pint) rather than naming the incompatibility between the two Quantities in question.

Reply:
> Quantity objects within Pandas DataFrames (or Series) will behave like Quantities, meaning that they are subject to unit conversion rules and will raise errors when incompatible units are mixed. But these loose Quantities don't offer the elegance or performance optimizations that come from using PintArrays. And they may give strange error messages as Pandas tries to convert incompatible units to dimensionless magnitudes (which is often prohibited by Pint) rather than naming the incompatibility between the two Quantities in question.

It took 2-3 reads for me to follow this, referring to quantity objects and loose quantities is ambiguous (could argue PintArray contains quantity objects since getitem returns them). Some code showing your points would be clearer. Can do that in another PR

Reply:
I think that the benchmark PR I made makes this point really clearly (up to 1000x performance difference).  So if/when that lands, we can refer to coding (and performance examples) from that.